sympy/integrals/transforms.py

```diff
@@ -1186,607 +1172,549 @@ def _laplace_build_rules(t, s):
+    for p_rule in prog_rules:
+        if (L := p_rule(f, t, s)) is not None:
+            return L
+    return None
 
+def _laplace_apply_simple_rules(f, t, s):
+    """
+    This function applies all simple rules and returns the result if one
+    of them gives a result.
+    """
     simple_rules = _laplace_build_rules(t, s)
+    prep_old = ''
+    prep_f = ''
     for t_dom, s_dom, check, plane, prep in simple_rules:
-        ma = prep(func).match(t_dom)
+        if not prep_old==prep:
+            prep_f = prep(f)
+            prep_old = prep
+        ma = prep_f.match(t_dom)
         if ma:
-            debug('_laplace_apply_rules match:')
-            debug('      f:    %s'%(func,))
-            debug('      rule: %s o---o %s'%(t_dom, s_dom))
             try:
-                debug('      try   %s'%(check,))
                 c = check.xreplace(ma)
-                debug('      check %s -> %s'%(check, c))
-                if c==True:
-                    return _laplace_cr(k*s_dom.xreplace(ma),
-                                plane.xreplace(ma), S.true, **hints)
-            except Exception:
-                debug('_laplace_apply_rules did not match.')
-    if f.has(DiracDelta):
-        return None
-
-    prog_rules = [_laplace_rule_timescale, _laplace_rule_heaviside,
-                  _laplace_rule_exp, _laplace_rule_trig, _laplace_rule_diff]
-    for p_rule in prog_rules:
-        L = p_rule(func, t, s, doit=doit, **hints)
-        if L is not None:
-            noconds = hints.get('noconds', False)
-            if not noconds and type(L) is tuple:
-                r, p, c = L
-                return (k*r, p, c)
-            else:
-                return k*L
+            except TypeError:
+                # This may happen if the time function has imaginary
+                # numbers in it. Then we give up.
+                continue
+            if c==True:
+                debug('_laplace_apply_simple_rules match:')
+                debug('      f:     %s'%(f,))
+                debug('      rule:  %s o---o %s'%(t_dom, s_dom))
+                debug('      match: %s'%(ma, ))
+                return (s_dom.xreplace(ma), plane.xreplace(ma), c)
     return None
 
+def _laplace_transform(fn, t_, s_, simplify=True):
+    """
+    Front-end function of the Laplace transform. It tries to apply all known
+    rules recursively, and if everything else fails, it tries to integrate.
+    """
+    debug('[LT _l_t] (%s, %s, %s)'%(fn, t_, s_))
+
+    terms = Add.make_args(fn)
+    results = []
+    for ff in terms:
+        k, ft = ff.as_independent(t_, as_Add=False)
+        if (r := _laplace_apply_simple_rules(ft, t_, s_)) is not None:
+            pass
+        elif (r := _laplace_apply_prog_rules(ft, t_, s_)) is not None:
+            pass
+        elif (r := _laplace_expand(ft, t_, s_)) is not None:
+            pass
+        elif ft.has(AppliedUndef):
+            r = (LaplaceTransform(ft, t_, s_), S.NegativeInfinity, True)
```

Comment:
Checking only `has(AppliedUndef)` like this will also catch cases where there is a function of a different variable like `laplace_transform(g(x)*sin(t), t, s)`.

Reply:
Yes, but such a thing cannot get to that point in the algorithm since it will be treated as a constant by everything else in front of it.

Reply:
There are many other ways like:
```
In [8]: sin(g(x)*t)
Out[8]: sin(t⋅g(x))
```

Reply:
I'll still change it, though